File: fjt.pricingservice/MySqlDBModel/rfq_lineitems.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fjt.pricingservice.MySqlDBModel
{
    [Table("rfq_lineitems")]
   public partial class rfq_lineitems
    {
        public rfq_lineitems()
        {   
            rfq_lineitems_alternatepart = new HashSet<rfq_lineitems_alternatepart>();
        }
        [Key]
        public int id { get; set; }
        public int? rfqAssyBomID { get; set; }
        public int? rfqAssyID { get; set; }
        [StringLength(8)]
        public string lineID { get; set; }
        public int? qpa { get; set; }
        [StringLength(500)]
        public string refDesig { get; set; }
       
        [StringLength(150)]
        public string custPN { get; set; }
        public int? custPNID { get; set; }
        public int? uomID { get; set; }
        [StringLength(500)]
        public string description { get; set; }
        public double? level { get; set; }
        public bool? isInstall { get; set; }
        public bool? isPurchase { get; set; }
        public bool? isNoBidsPN { get; set; }
        public bool? isDraft { get; set; }
        public bool? isActive { get; set; }
        public int? partTypeID { get; set; }
        public int? partclassID { get; set; }
        public decimal? leadQty { get; set; }
        public decimal? attritionRate { get; set; }
        public decimal? totalQty { get; set; }
        public decimal? manualAdj { get; set; }
        public bool isDeleted { get; set; }
        [StringLength(255)]
        public string createdBy { get; set; }
        public DateTime createdAt { get; set; }
        [StringLength(255)]
        public string updatedBy { get; set; }
        public DateTime? updatedAt { get; set; }
        [StringLength(255)]
        public string deletedBy { get; set; }
        public DateTime? deletedAt { get; set; }
        [StringLength(2)]
        public string customerRev { get; set; }
        [StringLength(2)]
        public string flextronRev { get; set; }
        [StringLength(500)]
        public string customerDescription { get; set; }
        public decimal? numOfPosition { get; set; }
        public decimal? valueAddedCost { get; set; }
        public int? refRFQLineItemID { get; set; }
        public int? dnpQty { get; set; }
        public int? dnpDesig { get; set; }
        [StringLength(8)]
        public string org_lineID { get; set; }
        public int? org_qpa { get; set; }
        [StringLength(500)]
        public string org_refDesig { get; set; }
        [StringLength(150)]
        public string org_custPN { get; set; }
        [StringLength(100)]
        public string org_uomName { get; set; }
        public double? org_level { get; set; }
        public bool? org_isInstall { get; set; }
        public bool? org_isPurchase { get; set; }
        [StringLength(2)]
        public string org_customerRev { get; set; }
        [StringLength(500)]
        public string org_customerDescription { get; set; }
        public decimal? org_numOfPosition { get; set; }
        public string org_refLineID { get; set; }
        public bool? qpaDesignatorStep { get; set; }
        public virtual rfq_assemblies rfq_assemblies { get; set; }
        public virtual rfq_assy_bom rfq_assy_bom { get; set; }
        public virtual ICollection<rfq_lineitems_alternatepart> rfq_lineitems_alternatepart { get; set; }
    }
}

File: fjt.api/models/RFQLineItems.js
const Sequelize = require('sequelize');

module.exports = (sequelize) => {
    const RFQLineItems = sequelize.define('RFQLineItems', {
        id: {
            allowNull: false,
            autoIncrement: true,
            primaryKey: true,
            type: Sequelize.INTEGER
        },
        lineID: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        qpa: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        refDesig: {
            type: Sequelize.STRING,
            allowNull: true
        },
        custPN: {
            type: Sequelize.STRING,
            allowNull: true
        },
        custPNID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        uomID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        description: {
            type: Sequelize.TEXT,
            allowNull: true
        },
        isInstall: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        isPurchase: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        isDeleted: {
            type: Sequelize.BOOLEAN,
            allowNull: true,
            defaultValue: false
        },
        createdBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        updatedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        customerRev: {
            type: Sequelize.STRING,
            allowNull: true
        },
        customerDescription: {
            type: Sequelize.STRING,
            allowNull: true
        },
        numOfPosition: {
            type: Sequelize.DECIMAL(6, 6),
            allowNull: true
        },
        dnpQty: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        dnpDesig: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_lineID: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_qpa: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        org_refDesig: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_custPN: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_uomName: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_isInstall: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_isPurchase: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_customerRev: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_customerDescription: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_numOfPosition: {
            type: Sequelize.DECIMAL(6, 6),
            allowNull: true
        },
        qpaDesignatorStep: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        customerPartDesc: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_customerPartDesc: {
            type: Sequelize.STRING,
            allowNull: true
        },
        mergeLines: {
            type: Sequelize.STRING,
            allowNull: true
        },
        lineMergeStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        isBuyDNPQty: {
            type: Sequelize.STRING,
            allowNull: true
        },
        partID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        duplicateCPNStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        requireMountingTypeStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        requireFunctionalTypeStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        programingStatus: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        refDesigCount: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        isObsoleteLine: {
            type: Sequelize.BOOLEAN,
            allowNull: true,
            defaultValue: false
        },
        cust_lineID: {
            type: Sequelize.STRING,
            allowNull: true
        },
        customerApprovalForQPAREFDESStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true,
            defaultValue: true
        },
        customerApprovalForBuyStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true,
            defaultValue: true
        },
        customerApprovalForPopulateStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true,
            defaultValue: true
        },
        substitutesAllow: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        org_substitutesAllow: {
            type: Sequelize.STRING,
            allowNull: true
        },
        numOfRows: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        org_numOfRows: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        restrictCPNUseWithPermissionStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        restrictCPNUsePermanentlyStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        restrictCPNUseInBOMStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        customerApprovalCPN: {
            type: Sequelize.TEXT,
            allowNull: true
        },
        customerApprovalCPNBy: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        customerApprovalCPNDate: {
            type: Sequelize.DATE,
            allowNull: true
        },
        dnpDesigCount: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        org_dnpQty: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        org_dnpDesig: {
            type: Sequelize.STRING,
            allowNull: true
        },
        org_buyDNPQty: {
            type: Sequelize.STRING,
            allowNull: true
        },
        requireMountingTypeError: {
            type: Sequelize.STRING,
            allowNull: true
        },
        requireFunctionalTypeError: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        dnpQPARefDesStep: {
            type: Sequelize.INTEGER,
            allowNull: true,
            defaultValue: true
        },
        customerApprovalForDNPQPAREFDESStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true,
            defaultValue: true
        },
        customerApprovalForDNPBuyStep: {
            type: Sequelize.BOOLEAN,
            allowNull: true,
            defaultValue: true
        },
        createByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        updateByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        deleteByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        isNotRequiredKitAllocation: {
            type: Sequelize.BOOLEAN,
            allowNull: false,
            defaultValue: false
        },
        isSupplierToBuy: {
            type: Sequelize.BOOLEAN,
            allowNull: false,
            defaultValue: false
        },
        programmingMappingPendingRefdesCount: {
            type: Sequelize.INTEGER,
            allowNull: true,
            defaultValue: 0
        }
    }, {
        paranoid: true,
        tableName: 'rfq_lineitems'
    });

    RFQLineItems.associate = (models) => {
        RFQLineItems.hasMany(models.RFQLineitemsAlternatepart, {
            as: 'rfqLineitemsAlternetpart',
            foreignKey: 'rfqLineItemsID'
        });

        RFQLineItems.hasMany(models.RFQLineitemsAdditionalComment, {
            as: 'rfqLineitemsAddtionalComment',
            foreignKey: 'rfqLineItemID'
        });

        RFQLineItems.belongsTo(models.Component, {
            as: 'custPNIDcomponent',
            foreignKey: 'custPNID'
        });

        RFQLineItems.belongsTo(models.UOMs, {
            as: 'uoms',
            foreignKey: 'uomID'
        });

        RFQLineItems.hasMany(models.ComponentCustomerLOA, {
            as: 'customerLOA',
            foreignKey: 'refLineitemID'
        });

        RFQLineItems.hasOne(models.RFQConsolidatedMFGPNLineItem, {
            as: 'rfqConsolidatedMFGPNLineItem',
            foreignKey: 'rfqLineItemID'
        });

        RFQLineItems.belongsTo(models.Component, {
            as: 'component',
            foreignKey: 'partID'
        });
        RFQLineItems.hasMany(models.PurchasePartsDetails, {
            as: 'PurchasePartsDetails',
            foreignKey: 'refBOMLineID'
        });
        RFQLineItems.hasMany(models.RFQLineItemsProgrammingMapping, {
            foreignKey: 'softwareRFQLineItemID',
            as: 'softwareLineItemsProgrammingMapping'
        });
        RFQLineItems.hasMany(models.RFQLineItemsProgrammingMapping, {
            foreignKey: 'rfqLineItemID',
            as: 'partLineItemsProgrammingMapping'
        });
    };

    return RFQLineItems;
};
